lib/tensor.ts

```diff
@@ -169,7 +168,8 @@ export class Tensor {
       /**
        * get the data ID that used to map to a tensor data
        */
-      public readonly dataId: Tensor.Id = {}) {
+      public readonly dataId: number = globalId) {
+    globalId += 1;
```

Comment:
do we need synchronization here?

Reply:
not if browser tab is using single thread -- and it seems it is the case.